### PR TITLE
CurrencyIn: Switch from Text File to JSON

### LINKED ISSUE #1616: CurrencyIn: Cleanup currencies.txt data file
The CurrencyIn Goodies has now been converted to use a structured answer, however the data file it uses could use some cleanup. Most of the currencies don't have they name of the currency properly capitalized (i.e. `Canadian dollar (CAD)` should be `Canadian Dollar (CAD)`).

IMO we should really switch from a simple text file to something more structured like JSON or YAML which is more common in the Goodies Repo. That would also make the data easier to update, and read. The code would need to be updated of course to handle the changes.

-----
https://duck.co/ia/view/currency_in

### PR DESCRIPTION
**Pull request by ozdemirburak**
Fixes #1616.

-----
IA Page: https://duck.co/ia/view/currency_in

### CODE REVIEW

**Comment by moollaza** [comment]
@ozdemirburak thanks! We'll take a look and review shortly. I've set it up for deployment to Beta.duckduckgo.com

File: lib/DDG/Goodie/CurrencyIn.pm [hunk 1]
```diff
@@ -62,33 +61,22 @@ handle remainder => sub {
...
         }
 
-        if (exists $countries{$country."\n"}){
-            my $string_currency = $countries{$country."\n"};    # Load currencies as string (one line from .txt)
-            my @currencies =  split(',', decode("utf8", $string_currency)); # Split currencies into array
-
-            my $count = $#currencies + 1;                       # Get number of currencies
-            my $output_country = $country;                      # Pass country name to the output_country
-            $output_country =~ s/\b(\w)/\U$1/g;                 # so it can by capitalized
+        if (exists $countries->{$country}){
```
Comment (moollaza) [inline lib/DDG/Goodie/CurrencyIn.pm]: While we're working on this, we might as well improve the code. Let's reduce this nesting and make the code a little more Perl-y: `return unless exists $countries->{$country};`

**Comment by moollaza** [comment]
@ozdemirburak one minor fix otherwise this LGTM :+1:

**Comment by ozdemirburak** [comment]
@moollaza Thanks, updated.

**Comment by moollaza** [comment]
@ozdemirburak Thanks! This LGTM :+1:

**Pull request merged**